### ISSUE TITLE
Checkout confilict with no changes

### ISSUE THREAD

**Issue by renewsorensen** [libgit2/libgit2sharp#1282]
I am facing a rather strange issue, it maybe that i am doing something wrong.

I am looping over all branches, i then locate certain files, i then read something from the files and then move on. But i can't move on because i get conflicts when i switch, and this is where it gets strange, because i do not change anything at anytime in this process, so why is it telling me that there is a conflict??

One thing i notices is that, when submodules is behind, don't know if that has something to do with it

`
var repo = new Repository("path to repo");
foreach (var branch in repo.Branches)
{
        repo.Checkout(branch);
        dosomthing(branch);
}
`

**Comment by ethomson** [comment]
Yes, that's likely the problem.  This sounds like the problem that https://github.com/libgit2/libgit2/pull/3583/files is addressing.

**Comment by renewsorensen** [comment]
I just tried to stage those submodules before i do the checkout, but that only sometimes i still get a conflict error. When i retrive the status after a stage a submodule it is not reflected in the status object?

`
   
 public void Switch(Branch branch)
    {
      var status = _repository.RetrieveStatus();

      foreach (var item in status)
      {
        if (item.FilePath == "SharedComponent" && item.State == FileStatus.ModifiedInWorkdir)
          _repository.Stage("SharedComponent");

        if (item.FilePath == @"MSBuild\BuildingEnvironment" && item.State == FileStatus.ModifiedInWorkdir)
          _repository.Stage(@"MSBuild\BuildingEnvironment");
      }


      if (_repository.RetrieveStatus().Any(s => s.State == FileStatus.ModifiedInIndex))
        _repository.Commit("align sub modules", new Signature("mr jenkins", "mrjenkins@domain.com", DateTime.Now), new CommitOptions() { AllowEmptyCommit = true });

      _repository.Checkout(branch.FriendlyName, new CheckoutOptions() { });
    }
`
This is status after i stage a submodule
Added: Count = 0
    added: Count = 0
    DebuggerDisplay: "+0 ~0 -0 | +0 ~1 -0 | i1"
    dispatcher: Count = 9
    Ignored: Count = 1
    ignored: Count = 1
    IsDirty: true
    isDirty: true
    Missing: Count = 0
    missing: Count = 0
    modified: Count = 1
    Modified: Count = 1
    Removed: Count = 0
    removed: Count = 0
    RenamedInIndex: Count = 0
    renamedInIndex: Count = 0
    renamedInWorkDir: Count = 0
    RenamedInWorkDir: Count = 0
    staged: Count = 0
    Staged: Count = 0
    statusEntries: Count = 2
    Unaltered: Count = 0
    unaltered: Count = 0
    untracked: Count = 0
    Untracked: Count = 0
